subprojects/core/src/main/java/org/gradle/api/internal/provider/DefaultProviderFactory.java

```diff
@@ -69,9 +67,9 @@ public DefaultProviderFactory(FileOperations fileOperations) {
             ((PropertyState<Double>) propertyState).set(Double.valueOf(0));
         } else if (clazz == Character.class) {
             ((PropertyState<Character>) propertyState).set(new Character('\0'));
-        } else if (clazz == FileCollection.class || clazz == ConfigurableFileCollection.class) {
+        } else if (clazz == ConfigurableFileCollection.class) {
```

Comment:
That's right. This is why we shouldn't be dealing with `FileCollection` and friends here, as it would be removed later. We should just not add it in the first place.

People can use `Provider<FileCollection>` if they really, really want to, as can the tests. However, it would be better for the tests to use some a provider of some scalar type, say `String` or `File`, as this is the use case we're starting with.